### ISSUE TITLE
Displaying path names in Windows format is awkward for bash users

### ISSUE THREAD

**Issue by jfehrle** [coq/coq#15859]
On Windows, displayed file pathnames are always shown in Windows pathname format.  But bash commands in cygwin don't understand windows pathnames.  The backslashes are not recognized as directory separators. (`C:/` works, though.)

`cygpath` can convert from one form to another.

It's unclear to me whether we should support running Coq on Windows outside of bash.

There are a couple ways we could address this:
- Change how we print paths on Windows and instruct Windows users to work in bash in the documentation.  Perhaps we could use `cygpath` to get the directory where Coq is installed, then create our own routine to format the path.  (My recommendation)
- If we want to support running Coq outside of bash, make the pathname format configurable on Windows (or detect whether the command was run from bash based if that's possible, e.g. from the command line arguments)
- Simply document the issue and mention the existence of `cygpath` (ugh)

For example: (Another place where you see Windows paths is in the output of the `Print LoadPath` command.)

```
$ coqc -where
C:\cygwin64_coq_platform\home\jfehr\.opam\__coq-platform.2022.01.0~8.15~preview1\lib/coq

$ ls C:\cygwin64_coq_platform\home\jfehr\.opam\__coq-platform.2022.01.0~8.15~preview1\lib/coq
ls: cannot access 'C:cygwin64_coq_platformhomejfehr.opam__coq-platform.2022.01.0~8.15~preview1lib/coq': No such file or directory

$ cygpath -u `coqc -where`
/home/jfehr/.opam/__coq-platform.2022.01.0~8.15~preview1/lib/coq

$ ls /home/jfehr/.opam/__coq-platform.2022.01.0~8.15~preview1/lib/coq
theories  user-contrib

$ ls C:/
'$Recycle.Bin'   Coq-debugger_preview     cygwin64_coq3             hp             PerfLogs               swapfile.sys
    :
```

**Comment by jfehrle** [comment]
@MSoegtropIMC Any thoughts?

**Comment by MSoegtropIMC** [comment]
I don't think it makes sense to change this, since coqc is a MinGW app and this is its native path format. There is hope that with dune one soon won't need a bash any more to build Coq projects.

You can easily use MinGW paths in a cygwin bash by quoting the complete path with ''. Cagwin actually does understand Windows path, one just has to quote the \.

I have no issue typing ' + Ctrl+V + ' when copy and pasting paths.

It might make more sense to add back cygwin as target (in addition to MinGW). The reason this does not exist is that OCaml was severely broken for Cygwin for a long time. I don't know if this is fixed meanwhile. One could give it a try. The scripts should be easy to patch.

**Comment by jfehrle** [comment]
You may be right.

I wanted to do ```ls `coqc -where` ```, but I think there's no way to do this in Windows in a single command.

**Comment by MSoegtropIMC** [comment]
I think
```
ls "`coqc -where`"
```
should do it (or the IMHO more readable `ls "$(coqc -where)"`).

**Comment by MSoegtropIMC** [comment]
I don't have time right now to test "native cygwin", but if you test it successfully, I would add it. The main question is if you can install ocaml via opam on cygwin, which one can test without patching the scripts.

**Comment by jfehrle** [comment]
Neither of these works, alas.  Not sure how it's getting the ```$`\r``` at the end.

I'm not going out of my way to use cygwin--I used the Coq platform script to install Coq.  That installs Cygwin, which knows where to find `coqc`, so I run it from cygwin bash.  My Windows command line is not configured to know where to find `coqc`.  Maybe I skipped a step in the installation as I had no intention of using the Windows commands line.

```
$ ls "`coqc -where`"
ls: cannot access 'C:\cygwin64_coq_platform\home\jfehr\.opam\__coq-platform.2022.01.0~8.15~preview1\lib/coq'$'\r': No such file or directory

$ ls "$(coqc -where)"
ls: cannot access 'C:\cygwin64_coq_platform\home\jfehr\.opam\__coq-platform.2022.01.0~8.15~preview1\lib/coq'$'\r': No such file or directory
```

**Comment by MSoegtropIMC** [comment]
Ah ok, that's because coqc is a MinGW application and outputs `\r\n` as line end. For cygwin the `\r` is a normal character, so it thinks it is part of the path. The $ is a prefix for ' ' to interpret backslash escaped characters (it usually doesn't). So '\r' is "backslash + r", while $'\r' is "carriage return".

It might make sense to add an option to coqc to use Linux line endings - or just make this the default even on Windows. But this would likely be a rather invasive change.

This should work:
```
ls "$(coqc -where | tr -d '\r')"
```
Note that tr itself intreprets the '\r', so no need for a $.
